File: http-process/src/main/java/com/github/dragonhht/http/response/Response.java
package com.github.dragonhht.http.response;

import javax.servlet.ServletOutputStream;
import javax.servlet.http.Cookie;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import java.io.*;
import java.util.Collection;
import java.util.Locale;

/**
 * Response类.
 *
 * @author: huang
 * Date: 2018/4/20
 */
final class Response implements HttpServletResponse {

    /** 静态资源文件存放路径. */
    private static final String WEB_ROOT = "WebRoot";

    private OutputStream out;
    private HttpServletRequest request;
    private PrintWriter writer;

    public Response(OutputStream out) {
        this.out = out;
    }

    public void setRequest(HttpServletRequest request) {
        this.request = request;
    }

    public void senStaticResource(String path) {
        StringBuffer outStr = new StringBuffer();
        outStr.append("HTTP/1.1 200 OK\n");
        outStr.append("Content-Type: text/html;charset=UTF-8\n");
        outStr.append("\r\n");
        try {
            out.write(outStr.toString().getBytes());
            getFile(path);
            out.flush();
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    /**
     * 获取静态文件.
     * @param path 静态文件路径
     * @throws IOException
     */
    private void getFile(String path) {
        if (path == null) return;
        path = WEB_ROOT + path;
        FileInputStream fis = null;
        try {
            try {
                fis = new FileInputStream(path);

            } catch (FileNotFoundException e) {
                path = WEB_ROOT + "/html/error/404.html";
                fis = new FileInputStream(path);
            }

            int len = 0;
            byte[] buffer = new byte[1024];
            while ((len = fis.read(buffer)) != -1) {
                out.write(buffer, 0, len);
            }
        } catch (IOException e) {
            e.printStackTrace();
        } finally {
            if (fis != null) {
                try {
                    fis.close();
                } catch (IOException e) {
                    e.printStackTrace();
                }
            }
        }
    }

    @Override
    public String getCharacterEncoding() {
        return null;
    }

    @Override
    public String getContentType() {
        return null;
    }

    @Override
    public ServletOutputStream getOutputStream() throws IOException {
        return null;
    }

    @Override
    public PrintWriter getWriter() throws IOException {
        writer = new PrintWriter(out, true);
        return writer;
    }

    @Override
    public void setCharacterEncoding(String charset) {

    }

    @Override
    public void setContentLength(int len) {

    }

    @Override
    public void setContentLengthLong(long len) {

    }

    @Override
    public void setContentType(String type) {

    }

    @Override
    public void setBufferSize(int size) {

    }

    @Override
    public int getBufferSize() {
        return 0;
    }

    @Override
    public void flushBuffer() throws IOException {

    }

    @Override
    public void resetBuffer() {

    }

    @Override
    public boolean isCommitted() {
        return false;
    }

    @Override
    public void reset() {

    }

    @Override
    public void setLocale(Locale loc) {

    }

    @Override
    public Locale getLocale() {
        return null;
    }

    @Override
    public void addCookie(Cookie cookie) {

    }

    @Override
    public boolean containsHeader(String s) {
        return false;
    }

    @Override
    public String encodeURL(String s) {
        return null;
    }

    @Override
    public String encodeRedirectURL(String s) {
        return null;
    }

    @Override
    public String encodeUrl(String s) {
        return null;
    }

    @Override
    public String encodeRedirectUrl(String s) {
        return null;
    }

    @Override
    public void sendError(int i, String s) throws IOException {

    }

    @Override
    public void sendError(int i) throws IOException {

    }

    @Override
    public void sendRedirect(String s) throws IOException {

    }

    @Override
    public void setDateHeader(String s, long l) {

    }

    @Override
    public void addDateHeader(String s, long l) {

    }

    @Override
    public void setHeader(String s, String s1) {

    }

    @Override
    public void addHeader(String s, String s1) {

    }

    @Override
    public void setIntHeader(String s, int i) {

    }

    @Override
    public void addIntHeader(String s, int i) {

    }

    @Override
    public void setStatus(int i) {

    }

    @Override
    public void setStatus(int i, String s) {

    }

    @Override
    public int getStatus() {
        return 0;
    }

    @Override
    public String getHeader(String s) {
        return null;
    }

    @Override
    public Collection<String> getHeaders(String s) {
        return null;
    }

    @Override
    public Collection<String> getHeaderNames() {
        return null;
    }
}

File: netty-server/src/main/java/com/github/dragonhht/http/servlet/Response.kt
package com.github.dragonhht.http.servlet

import java.io.IOException
import java.io.OutputStream
import java.io.PrintWriter
import java.util.*
import javax.servlet.ServletOutputStream
import javax.servlet.http.Cookie
import javax.servlet.http.HttpServletResponse

/**
 * .
 *
 * @author: huang
 * @Date: 2019-4-17
 */
class Response constructor(var out: OutputStream): HttpServletResponse {
    private var writer: PrintWriter? = null
    private var servletOutputStream: ServletOutputStream? = null

    override fun getCharacterEncoding(): String? {
        return null
    }

    override fun getContentType(): String? {
        return null
    }

    @Throws(IOException::class)
    override fun getOutputStream(): ServletOutputStream? {
        if (this.servletOutputStream != null) {
            return this.servletOutputStream
        }
        this.servletOutputStream = ResponseOutputStream(this.out)
        return this.servletOutputStream
    }

    @Throws(IOException::class)
    override fun getWriter(): PrintWriter? {
        writer = PrintWriter(out, true)
        return writer
    }

    override fun setCharacterEncoding(charset: String) {

    }

    override fun setContentLength(len: Int) {

    }

    override fun setContentLengthLong(len: Long) {

    }

    override fun setContentType(type: String) {

    }

    override fun setBufferSize(size: Int) {

    }

    override fun getBufferSize(): Int {
        return 0
    }

    @Throws(IOException::class)
    override fun flushBuffer() {

    }

    override fun resetBuffer() {

    }

    override fun isCommitted(): Boolean {
        return false
    }

    override fun reset() {

    }

    override fun setLocale(loc: Locale) {

    }

    override fun getLocale(): Locale? {
        return null
    }

    override fun addCookie(cookie: Cookie) {

    }

    override fun containsHeader(s: String): Boolean {
        return false
    }

    override fun encodeURL(s: String): String? {
        return null
    }

    override fun encodeRedirectURL(s: String): String? {
        return null
    }

    override fun encodeUrl(s: String): String? {
        return null
    }

    override fun encodeRedirectUrl(s: String): String? {
        return null
    }

    @Throws(IOException::class)
    override fun sendError(i: Int, s: String) {

    }

    @Throws(IOException::class)
    override fun sendError(i: Int) {

    }

    @Throws(IOException::class)
    override fun sendRedirect(s: String) {

    }

    override fun setDateHeader(s: String, l: Long) {

    }

    override fun addDateHeader(s: String, l: Long) {

    }

    override fun setHeader(s: String, s1: String) {

    }

    override fun addHeader(s: String, s1: String) {

    }

    override fun setIntHeader(s: String, i: Int) {

    }

    override fun addIntHeader(s: String, i: Int) {

    }

    override fun setStatus(i: Int) {

    }

    override fun setStatus(i: Int, s: String) {

    }

    override fun getStatus(): Int {
        return 0
    }

    override fun getHeader(s: String): String? {
        return null
    }

    override fun getHeaders(s: String): Collection<String>? {
        return null
    }

    override fun getHeaderNames(): Collection<String>? {
        return null
    }
}